sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -661,6 +663,68 @@ public CosmosAsyncConflict getConflict(String id) {
+                                                , new FeedOptions())
+                                            .single()
+                                            .flatMap(offerFeedResponse -> {
+                                                if (offerFeedResponse.getResults().isEmpty()) {
+                                                    return Mono.error(BridgeInternal
+                                                                          .createCosmosClientException(
+                                                                              HttpConstants.StatusCodes.BADREQUEST,
+                                                                              "No offers found for the " +
+                                                                                  "resource"));
+                                                }
+
+                                                Offer existingOffer = offerFeedResponse.getResults().get(0);
+                                                ModelBridgeInternal.updateOfferFromProperties(existingOffer,
+                                                                                              throughputProperties);
+                                                return this.database.getDocClientWrapper()
+                                                           .replaceOffer(existingOffer)
+                                                           .single();
+                                            }).map(ModelBridgeInternal::createThroughputRespose));
+    }
+
+    /**
+     * Read the throughput throughput .
+     *
+     * @return the mono containing throughput response
+     */
+    public Mono<ThroughputResponse> readThroughput() {
+        return this.read()
+                   .flatMap(response -> this.database.getDocClientWrapper()
+                                            .queryOffers(database.getOfferQuerySpecFromResourceId(response
+                                                                                                      .getProperties()
+                                                                                                      .getResourceId())
+                                                , new FeedOptions())
+                                            .single()
+                                            .flatMap(offerFeedResponse -> {
+                                                if (offerFeedResponse.getResults().isEmpty()) {
+                                                    return Mono.error(BridgeInternal
+                                                                          .createCosmosClientException(
+                                                                              HttpConstants.StatusCodes.BADREQUEST,
+                                                                              "No offers found for the resource"));
```

Comment:
it is worth including the name of the database and container in the sdk generated error message.